src/averageAges.js

```diff
@@ -56,6 +76,21 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+
+  const mothers = people.filter(person => person.sex === 'f'
+    && people.some(women => women.mother === person.name));
+
+  const children = onlyWithSon ? people
+    .filter(person => mothers.some(mother => mother.name === person.mother)
+    && person.sex === 'm') : people
+    .filter(person => mothers.some(mother => mother.name === person.mother));
+
+  const ageDifference = children.reduce((a, child) => a
```

Comment:
`a` that's a bad name for the variable

Reply:
use meaningful names

Reply:
"a" is a short way to write accumulator as we use "i" for index, but maybe you got a point.

Reply:
`i` inside `for` loop that's an exception, when you use methods like `reduce` you want to know what it is, so it's better to write accumulator if you mean accumulator